src/basic/fileio.c

```diff
@@ -71,7 +71,7 @@ int write_string_stream_ts(FILE *f, const char *line, bool enforce_newline, stru
         return fflush_and_check(f);
 }
 
-static int write_string_file_atomic(const char *fn, const char *line, bool enforce_newline) {
+static int write_string_file_atomic(const char *fn, const char *line, bool enforce_newline, bool sync) {
```

Comment:
glibc exports a function called sync(), we should probably avoid overloading that as a variable here locally (gcc even used to warn about that, not sure why it doesn't anymore), to avoid confusion around what "if (sync)" actually means...

I wonder if it wouldn't be easier to read to just change this call (and related ones) to just take a WriteStringFileFlags, to unify behaviour here... Functions with many bool params are so hard to call, since you never know which bool param you are actually passing there...

Reply:
Well spotted.

bool parameters are horrible, but I think they work in this case.  It can only be called locally, and both callers use a syntax which has fairly clear intent.  I think it's clearer than having to work out which functions are responsible for which of the flags.  Maybe commenting the function as such and masking the relevant flags would make it ok, but I'm not sure how much that helps.